File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.c
int maxProfit(int* prices, int pricesSize) {
    int flag =0;
     if (pricesSize <= 1) return 0;
     int l=0,r=0,max=0,count=0;
    while(r<pricesSize && l<pricesSize){
        if((prices[r]-prices[l])>0){
            max=(prices[r]-prices[l]);
            if(max>count) count=max;
        }
        else{
            l=r;
        }
        r++;
    }
    return count;
    
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: list[int]) -> int:
        n=len(prices)
        arr=[0]*n
        ans=0
        maxi=0
        for i in range(n-1,-1,-1):
            maxi=max(maxi,prices[i])
            arr[i]=maxi
        """     i=0
        for val in prices:
            ans=max(ans,arr[i]-val)
            i+=1
        """
        for i,val in enumerate(prices):
            ans=max(ans,arr[i]-val)
        return ans